### PR TITLE
check_isa.cpp, ispc.cpp: do not assume "not arm" means "x86_64"

### PR DESCRIPTION
**Pull request by goshhhy**
currently, if you attempt to compile ispc on a host which is not x86_64 or arm, the build fails on inline assembly directives, with the reason why being potentially non-obvious. for example:

```
linear@mihari:~/code/ispc/build $ uname -m
ppc64le
linear@mihari:~/code/ispc/build $ make
[  1%] Building CXX object CMakeFiles/check_isa.dir/check_isa.cpp.o
/usr/people/linear/code/ispc/check_isa.cpp:25:36: error: invalid output constraint '=a' in asm
    __asm__ __volatile__("cpuid" : "=a"(info[0]), "=b"(info[1]), "=c"(info[2]), "=d"(info[3]) : "0"(infoType));
                                   ^
/usr/people/linear/code/ispc/check_isa.cpp:29:36: error: invalid output constraint '=a' in asm
    __asm__ __volatile__("cpuid" : "=a"(info[0]), "=b"(info[1]), "=c"(info[2]), "=d"(info[3]) : "0"(level), "2"(count));
                                   ^
/usr/people/linear/code/ispc/check_isa.cpp:43:53: error: invalid output constraint '=a' in asm
    __asm__ __volatile__(".byte 0x0f, 0x01, 0xd0" : "=a"(rEAX), "=d"(rEDX) : "c"(0));
                                                    ^
/usr/people/linear/code/ispc/check_isa.cpp:69:53: error: invalid output constraint '=a' in asm
    __asm__ __volatile__(".byte 0x0f, 0x01, 0xd0" : "=a"(rEAX), "=d"(rEDX) : "c"(0));
                                                    ^
/usr/people/linear/code/ispc/check_isa.cpp:84:53: error: invalid output constraint '=a' in asm
    __asm__ __volatile__(".byte 0x0f, 0x01, 0xd0" : "=a"(rEAX), "=d"(rEDX) : "c"(0));
                                                    ^
5 errors generated.
make[2]: *** [CMakeFiles/check_isa.dir/build.make:76: CMakeFiles/check_isa.dir/check_isa.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:174: CMakeFiles/check_isa.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
linear@mihari:~/code/ispc/build $
```

this is because the code currently assumes that an architecture which is not arm, must be x86_64.

this pull request makes changes to check_isa.cpp and src/ispc.cpp to explicitly check for supported architectures instead. and adding a more informative error if an unsupported host architecture is in use. it also renames the definition used in ispc.cpp to align with those used for operating system detection. after this change, attempting to build on an unsupported platform looks like this instead:
```
linear@mihari:~/code/ispc/build $ uname -m
ppc64le
linear@mihari:~/code/ispc/build $ make
[  1%] Building CXX object CMakeFiles/check_isa.dir/check_isa.cpp.o
/usr/people/linear/code/ispc/check_isa.cpp:203:2: error: "Unsupported host CPU architecture."
#error "Unsupported host CPU architecture."
 ^
1 error generated.
make[2]: *** [CMakeFiles/check_isa.dir/build.make:76: CMakeFiles/check_isa.dir/check_isa.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:174: CMakeFiles/check_isa.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
linear@mihari:~/code/ispc/build $
```

### CODE REVIEW

File: check_isa.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #define HOST_IS_APPLE
 #endif
 
-#if !defined(__arm__) && !defined(__aarch64__) && !defined(_M_ARM64)
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_AMD64)
```
Comment (dbabokin) [inline check_isa.cpp:22]: `_M_AMD64` and `_M_X64` are equivalent according to this [blog](https://learn.microsoft.com/en-us/archive/blogs/reiley/macro-revisited). I would prefer `_M_X64`, it's just more clear naming.

**Review by dbabokin — COMMENTED** [review]
I generally like the changes, with minor remark. I see that you are trying to compile for PPC64 arch. But to truly work on this arch, new target will need to be introduced. Do you have plans to work on that?

**Comment by goshhhy** [comment]
> I generally like the changes, with minor remark. I see that you are trying to compile for PPC64 arch. But to truly work on this arch, new target will need to be introduced. Do you have plans to work on that?

i am considering doing this, but have not started in earnest. i see this change as an easy architecture-independent first step, since in addition to producing a better error message, it also simplifies adding a new architecture to this detection code.

**Review by nurmukhametov — APPROVED** [review]
LGTM!

File: check_isa.cpp (hunk 1 above)
Comment (nurmukhametov) [reply check_isa.cpp]: CI looks good

**Pull request merged**